### PR TITLE
ENH Adds no_batch_dim support for pad 2d and 3d

### PR DESCRIPTION
**Pull request by thomasjpfan**
Towards #60585

### CODE REVIEW

File: aten/src/ATen/native/cuda/ReflectionPad.cu [hunk 1]
```diff
@@ -580,8 +580,8 @@ TORCH_IMPL_FUNC(reflection_pad3d_out_cuda) (
         auto output_inner = output;
         if (!batch_mode) {
           // non-batch mode
-          auto input_inner = input.unsqueeze(0);
-          auto output_inner = output.unsqueeze(0);
+          input_inner = input.unsqueeze(0);
+          output_inner = output.unsqueeze(0);
```
Comment (thomasjpfan) [inline aten/src/ATen/native/cuda/ReflectionPad.cu:584]: This is a bug fix where the no_batch_dim test serves as a non-regression test.

File: aten/src/ATen/native/cuda/ReplicationPadding.cu [hunk 2]
```diff
@@ -673,8 +673,8 @@ TORCH_IMPL_FUNC(replication_pad3d_out_cuda) (
       at::Tensor input_ = input;
       at::Tensor output_ = output;
       if (numInputDims == 4) {
-        auto input_ = input.unsqueeze(0);
-        auto output_ = output.unsqueeze(0);
+        input_ = input.unsqueeze(0);
+        output_ = output.unsqueeze(0);
```
Comment (thomasjpfan) [inline aten/src/ATen/native/cuda/ReplicationPadding.cu:677]: Same here

File: aten/src/ATen/native/cuda/ReflectionPad.cu (hunk 1 above)
Comment (jbschlosser) [reply aten/src/ATen/native/cuda/ReflectionPad.cu:584]: Just to clarify, the bug kept non-batch mode from functioning properly?

**Review by jbschlosser — APPROVED** [review]
LGTM!

File: aten/src/ATen/native/cuda/ReflectionPad.cu (hunk 1 above)
Comment (thomasjpfan) [reply aten/src/ATen/native/cuda/ReflectionPad.cu:584]: > Just to clarify, the bug kept non-batch mode from functioning properly?

Yup. In this case, "non-batch mode" == "no batch dims". Now that we have better language for batch dims, I would want to renamed the `batch_mode` to `has_batch_dim`.

**Pull request closed without merging**